src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -1195,86 +1237,111 @@ int fx_muxer_t::read_config_and_execute(
-    }
+    auto app = new fx_definition_t();
+    fx_definitions_owned.push_back(std::unique_ptr<fx_definition_t>(app));
+    fx_definitions.push_back(app);
 
-    runtime_config_t config(config_file, dev_config_file);
-    if (!config.is_valid())
+    int rc = read_config(*app, app_candidate, runtime_config);
+    if (rc)
     {
-        trace::error(_X("Invalid runtimeconfig.json [%s] [%s]"), config.get_path().c_str(), config.get_dev_path().c_str());
-        return StatusCode::InvalidConfigFile;
+        return rc;
     }
 
-    // Append specified probe paths first and then config file probe paths into realpaths.
-    std::vector<pal::string_t> probe_realpaths;
-    for (const auto& path : spec_probe_paths)
-    {
-        append_probe_realpath(path, &probe_realpaths, config.get_tfm());
-    }
-    for (const auto& path : config.get_probe_paths())
-    {
-        append_probe_realpath(path, &probe_realpaths, config.get_tfm());
-    }
+    auto config = app->get_runtime_config();
 
     // 'Roll forward on no candidate fx' is disabled by default. It can be enabled through:
     // 1. Command line argument (--roll-forward-on-no-candidate-fx)
     // 2. Runtimeconfig json file ('rollForwardOnNoCandidateFx' property)
     // 3. DOTNET_ROLL_FORWARD_ON_NO_CANDIDATE_FX env var
     // The conflicts will be resolved by following the priority rank described above (from 1 to 3).
     // The env var condition is verified in the config file processing
-    int roll_fwd_on_no_candidate_fx_val = config.get_roll_fwd_on_no_candidate_fx();
     if (!roll_fwd_on_no_candidate_fx.empty())
     {
-        roll_fwd_on_no_candidate_fx_val = pal::xtoi(roll_fwd_on_no_candidate_fx.c_str());
+        config.set_get_roll_fwd_on_no_candidate_fx(pal::xtoi(roll_fwd_on_no_candidate_fx.c_str()));
     }
 
+    // Determine additional deps
     pal::string_t additional_deps_serialized;
-
     bool is_portable = config.get_portable();
-    if (is_portable) {
+    if (is_portable)
+    {
         additional_deps_serialized = additional_deps;
         if (additional_deps_serialized.empty())
         {
             // additional_deps_serialized stays empty if DOTNET_ADDITIONAL_DEPS env var is not defined
             pal::getenv(_X("DOTNET_ADDITIONAL_DEPS"), &additional_deps_serialized);
         }
+
+        // Obtain frameworks\platforms
+        auto version = fx_version_specified;
+        while (!config.get_fx_name().empty() && !config.get_fx_version().empty())
+        {
+            fx_definition_t* fx = resolve_fx_dir(mode, config, own_dir, version);
+            if (fx == nullptr)
+            {
+                pal::string_t searched_version = fx_version_specified.empty() ? config.get_fx_version() : fx_version_specified;
+                handle_missing_framework_error(mode, config.get_fx_name(), searched_version, pal::string_t(), own_dir);
+                return CoreHostLibMissingFailure;
```

Comment:
This isn't really the right return code, right?  We are probing other shared frameworks, which will never have `CoreHostLib`.  Can we define a new return code for this?

Reply:
Good point; note the existing code used this as well when the framework wasn't found (in addition to when the hostpolicy wasn't found). I added FrameworkMissingFailure value